### PR TITLE
chore: tidying up types

### PR DESCRIPTION
**Pull request by ignatiusmb**
A prelude before moving types into its own folder as mentioned in https://github.com/sveltejs/kit/pull/980#issuecomment-818492644. This PR actions consists of

- removing unused type introduced in https://github.com/sveltejs/kit/pull/901#discussion_r612271006
- syncing up handle hooks type docs
- merging unnecessary destructuring/duplication in types

No tests or changeset should be needed as this should not change anything

***

There's also a couple more things I'd like to do but not sure what the general opinion here is, this suggestions could be applied here or in a separate PR (or not at all)

1. only exported types from `types.d.ts` are exposed to the end-user
    a. it might be better to have not-`export`ed to be placed in other file (maybe `internal` or `config`), _or_
    b. just put it directly in the arguments since it's only used once, e.g. `AdapterUtils`
2. continuing 1b, types used only once like `ParameterizedBody` can be removed and placed directly (inline)

### CODE REVIEW

File: packages/kit/types.d.ts [hunk 1]
```diff
@@ -104,10 +104,7 @@ export type GetSession<Context = any, Session = any> = {
 	({ context }: { context: Context }): Session | Promise<Session>;
 };
 
-export type Handle<Context = any> = ({
-	request,
-	render
-}: {
+export type Handle<Context = any> = (handle: {
```
Comment (benmccann) [inline packages/kit/types.d.ts:107]: this is saying the argument to `Handle` is called `handle`?

File: documentation/docs/04-hooks.md [hunk 2]
```diff
@@ -95,10 +95,10 @@ type Response = {
 	body?: any;
 };
 
-type Handle<Context = any> = (
+type Handle<Context = any> = ({
```
Comment (benmccann) [inline documentation/docs/04-hooks.md:98]: thanks for fixing this! I noticed it the other day as well, but forgot to update

**Comment by benmccann** [comment]
lgtm except for maybe the `handle` name

File: packages/kit/types.d.ts (hunk 1 above)
Comment (ignatiusmb) [reply packages/kit/types.d.ts:107]: Yes, this also feels weird for me, I can't seem to find any mention about the actual object name in the docs too, only mentions "It receives the `request` object and `render` method, which calls SvelteKit's default renderer."

Maybe just a `input` or `incoming` like the others? 

https://github.com/sveltejs/kit/blob/11ec751b1158edf636b2ad28fd71706f43b62ec0/packages/kit/types.d.ts#L97-L101

**Review by dummdidumm — APPROVED** [review]
For the handle param name I suggest `input`

**Pull request merged**